lib/scout_apm/instant/middleware.rb

```diff
@@ -0,0 +1,116 @@
+module ScoutApm
+  module Instant
+
+    # an abstraction for manipulating the HTML we capture in the middleware
+    class Page
+      def initialize(html)
+        @html = html
+        @to_add_to_head = []
+        @to_add_to_body = []
+      end
+
+      def add_to_head(content)
+        @to_add_to_head << content
+      end
+
+      def add_to_body(content)
+        @to_add_to_body << content
+      end
+
+      def res
+        i = @html.index("</body>")
+        @html = @html.insert(i, @to_add_to_body.join("")) if i
+        i = @html.index("</head>")
+        @html = @html.insert(i, @to_add_to_head.join("")) if i
+        @html
+      end
+    end
+
+    class Util
+      # reads the literal contents of the file in assets/#{name}
+      # if any vars are supplied, do a simple string substitution of the vars for their values
+      def self.read_asset(name, vars = {})
+        contents = File.read(File.join(File.dirname(__FILE__), "assets", name))
+        if vars.any?
+          vars.each_pair{|k,v| contents.gsub!(k.to_s,v.to_s)}
+        end
+        contents
+      end
+    end
+
+    class Middleware
+      # def initialize(app)
+      #   @app = app
+      # end
+      #
+      # def call(env)
+      # end
+
+      def initialize(*args)
+        @args = args
+      end
+
+      def call(env)
+        "#{self.class}::Logic".constantize.new(*@args).call(env)
```

Comment:
It's forcing the middleware code to reload. I'll take this out before general release.